ocaml/idl/datamodel.ml

```diff
@@ -5206,6 +5211,30 @@ module SDN_controller = struct
       ~allowed_roles:_R_POOL_OP
       ()
 
+  let open_port = call
+      ~name:"open_port"
+      ~doc:"Open a port and add it to the list of opened ports"
+      ~params: [
+        Ref _sdn_controller, "self", "this SDN controller";
+        sdn_port_protocol, "protocol", "The protocol (tcp | udp)";
+        Int, "port", "The port to open"
```

Comment:
Can the port number be hardcoded instead? Adding the ability to open arbitrary ports won't sit well with the security team.

Reply:
I thought about, like having a map and specifying a key; for instance `openflow` -> `tcp:6653`.
But I'm not sure how to have an exhaustiv list of all the ports that could be used by a sdn controller.

Reply:
Perhaps it would also be useful to enable this only if a feature flag is enabled (see features.ml/features.mli). That way if the feature is not enabled you are not exposed to the security risks of having additional ports open.

Reply:
Indeed.